lab-elizabeth/app/entry.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+// require webpack assets
+require('./scss/main.scss');
+
+// npm modules
+const cowsay = require('cowsay-browser');
+const angular = require('angular');
+
+// app modules
+
+// angular module
+const demoApp = angular.module('demoApp', []);
+
+// angular constructions
+demoApp.controller('CowsayController', [ '$log', CowsayController]);
+
+function CowsayController($log){
+  $log.debug('init cowsayCtrl');
+  this.title = 'I am Cow!';
+  this.historyText = [];
+
+  cowsay.list((err, cowFiles) => {
+    this.cowFiles = cowFiles;
+    this.currentCow = this.cowFiles[0];
+  });
+
+  this.updateCow = function(input){
+    $log.debug('cowsayCtrl.updateCow()');
+    return '\n' + cowsay.say({text: input || 'gimme something to say!', f: this.currentCow});
+  };
+
+  this.speak = function(input){
+    $log.debug('this.updateCow()');
+    this.spoken = input;
+    if(input === ''){
+      this.spoken = 'gimme something to say!';
+    }
```

Comment:
Alternatively, you could account for either input or no input with short-circuiting like this:
```
this.spoken = input || 'gimme something to say'
```